packages/grid/_modules_/grid/components/column-header-item.tsx

```diff
@@ -8,131 +8,132 @@ import { classnames } from '../utils';
   options: GridOptions;
-  onColumnDragEnter?: (event: Event) => void;
-  onColumnDragOver?: (col: ColDef, coordinates: CursorCoordinates) => void;
-  onColumnDragStart?: (col: ColDef, currentTarget: HTMLElement) => void;
   separatorProps: React.HTMLAttributes<HTMLDivElement>;
 }
 
-export const ColumnHeaderItem = React.memo(
-  ({
-    column,
-    colIndex,
-    onColumnDragStart,
-    onColumnDragEnter,
-    onColumnDragOver,
-    isResizing,
-    separatorProps,
-    sortModel,
-    options,
-  }: ColumnHeaderItemProps) => {
-    const apiRef = React.useContext(ApiContext);
-    const { disableColumnReorder, showColumnRightBorder, disableColumnResize } = options;
+export const ColumnHeaderItem = ({
+  column,
+  isDragging,
+  colIndex,
+  isResizing,
+  separatorProps,
+  sortModel,
+  options,
+}: ColumnHeaderItemProps) => {
+  const apiRef = React.useContext(ApiContext);
+  const { disableColumnReorder, showColumnRightBorder, disableColumnResize } = options;
 
-    const columnSortModel = React.useMemo(
-      () =>
-        sortModel
-          .filter((model) => model.field === column.field)
-          .map((item) => ({
-            sortDirection: item.sort,
-            ...(sortModel.length <= 1 ? {} : { sortIndex: sortModel.indexOf(item) + 1 }),
-          }))[0] || {},
-      [column.field, sortModel],
-    );
+  const columnSortModel = React.useMemo(
+    () =>
+      sortModel
+        .filter((model) => model.field === column.field)
+        .map((item) => ({
+          sortDirection: item.sort,
+          ...(sortModel.length <= 1 ? {} : { sortIndex: sortModel.indexOf(item) + 1 }),
+        }))[0] || {},
+    [column.field, sortModel],
+  );
 
-    let headerComponent: React.ReactElement | null = null;
-    if (column.renderHeader) {
-      headerComponent = column.renderHeader({
-        api: apiRef!.current!,
-        colDef: column,
-        colIndex,
-        field: column.field,
-      });
-    }
+  let headerComponent: React.ReactElement | null = null;
+  if (column.renderHeader) {
+    headerComponent = column.renderHeader({
+      api: apiRef!.current!,
+      colDef: column,
+      colIndex,
+      field: column.field,
+    });
+  }
 
-    const dragConfig = {
-      draggable:
-        !disableColumnReorder && !!onColumnDragStart && !!onColumnDragEnter && !!onColumnDragOver,
-      onDragStart: onColumnDragStart && ((event) => onColumnDragStart(column, event.currentTarget)),
-      onDragEnter: onColumnDragEnter && ((event) => onColumnDragEnter(event)),
-      onDragOver:
-        onColumnDragOver &&
-        ((event) => {
-          onColumnDragOver(column, {
-            x: event.clientX,
-            y: event.clientY,
-          });
-        }),
-    };
-    const width = column.width!;
+  const dragConfig = {
+    draggable:
+      !disableColumnReorder &&
+      !!apiRef!.current.onColItemDragStart &&
+      !!apiRef!.current.onColItemDragEnter &&
+      !!apiRef!.current.onColItemDragOver,
```

Comment:
you don't need those functions checks as they will always be !!apiRef!.current.onColItemDragStart &&

Reply:
Well not necessarily, if you remove the `useColumnReorder` hook from the `GridContainer` component then this will fail, but if we assume that this feature will be there then yes they will be true. I'll remove them tho.

Reply:
Done